airbyte-cdk/python/airbyte_cdk/test/utils/reading.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from typing import Any, List, Mapping, Optional
+
+from airbyte_cdk import AbstractSource
+from airbyte_cdk.test.catalog_builder import CatalogBuilder
+from airbyte_cdk.test.entrypoint_wrapper import EntrypointOutput, read
+from airbyte_protocol.models import AirbyteStateMessage, ConfiguredAirbyteCatalog, SyncMode
+
+
+def catalog(stream_name: str, sync_mode: SyncMode) -> ConfiguredAirbyteCatalog:
+    """Create a catalog with a single stream."""
+    return CatalogBuilder().with_stream(stream_name, sync_mode).build()
+
+
+def read_records(
```

Comment:
I find this interface brittle. If at some point we need more than just the `stream_name, sync_mode` to describe the catalog, we will have to update this interface. Why not pass the catalog directly? We have a builder that should ease the process of instantiating it

Reply:
Same as the other comment above in `reading.py`